src/shared/Git-Credential-Manager/Program.cs

```diff
@@ -43,16 +43,24 @@ public static void Main(string[] args)
                 //
                 // On UNIX systems we do the same check, except instead of a copy we use a symlink.
                 //
-                string oldName = PlatformUtils.IsWindows()
-                    ? "git-credential-manager-core.exe"
-                    : "git-credential-manager-core";
 
-                if (appPath?.EndsWith(oldName, StringComparison.OrdinalIgnoreCase) ?? false)
+                if (!string.IsNullOrWhiteSpace(appPath))
                 {
-                    context.Streams.Error.WriteLine(
-                        "warning: git-credential-manager-core was renamed to git-credential-manager");
-                    context.Streams.Error.WriteLine(
-                        $"warning: see {Constants.HelpUrls.GcmExecRename} for more information");
+                    // Trim any (.exe) file extension if we're on Windows
+                    // Note that in some circumstances (like being called by Git when config is set
+                    // to just `helper = manager-core`) we don't always have ".exe" at the end.
+                    if (PlatformUtils.IsWindows() && appPath.EndsWith(".exe", StringComparison.OrdinalIgnoreCase))
+                    {
+                        appPath = appPath.Substring(0, appPath.Length - 4);
+                    }
+
+                    if (appPath.EndsWith("git-credential-manager-core", StringComparison.OrdinalIgnoreCase))
```

Comment:
I guess we do not need to make sure that there isn't any non-slash before `git-credential-manager-core`, right?

Reply:
If the user has a link or renamed executable like `foogit-credential-manager-core` then seeing an extra warning about the executable being named wrong.. is less an issue IMO :)
Otherwise I'd need to do extra checks for either `\git-cred..` or `/git-cred..`.